src/librustdoc/html/format.rs

```diff
@@ -613,7 +680,11 @@ pub(crate) fn href_with_root_path(
                     },
                 )
             } else {
-                return Err(HrefError::NotInExternalCache);
+                if matches!(def_kind, DefKind::Macro(_)) {
```

Comment:
This `if` can be joined with the `else` above it to make an `else if`.

Reply:
It's something I do often: keeping things "together" under a same "else" block. Not sure why. Well, fixed in any case. :)